Snippets/ASBFunctions/ASBFunctions_1.3/FunctionsHostBuilderUsage.cs

```diff
@@ -0,0 +1,141 @@
+{
+    #region asb-function-hostbuilder-trigger
+    // <autogenerated/>
+    using System.Threading.Tasks;
+    using Microsoft.Azure.ServiceBus;
+    using Microsoft.Azure.ServiceBus.Core;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+    using NServiceBus;
+
+    class FunctionEndpointTrigger
+    {
+        readonly IFunctionEndpoint endpoint;
+
+        public FunctionEndpointTrigger(IFunctionEndpoint endpoint)
+        {
+            this.endpoint = endpoint;
+        }
+
+        public async Task Run(
+            [ServiceBusTrigger(queueName: "ASBTriggerQueue")]
+            Message message,
+            ExecutionContext executionContext,
+            ILogger logger,
+            IMessageReceiver messageReceiver)
+        {
+            await endpoint.Process(message, executionContext, messageReceiver, logger);
+        }
+    }
+    #endregion
+}
+
+namespace ASBFunctions_1_3
+{
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+    using NServiceBus;
+
+    public class FunctionsHostBuilderUsage
+    {
+        class IConfigurationUsage
+        {
+            #region asb-function-default
+            class Startup : FunctionsStartup
+            {
+                public override void Configure(IFunctionsHostBuilder builder)
+                {
+                    builder.UseNServiceBus();
+                }
+            }
+            #endregion
+        }
+
+        #region asb-function-hostbuilder
+        class HostBuilderStartup
+        {
+            class Startup : FunctionsStartup
+            {
+                public override void Configure(IFunctionsHostBuilder builder)
+                {
+                    builder.UseNServiceBus(() =>
+                    {
+                        var configuration = new ServiceBusTriggeredEndpointConfiguration("MyFunctionsEndpoint");
+                        configuration.Transport.ConnectionString("functionConnectionString");
+                        return configuration;
+                    });
+                }
+            }
+        }
+        #endregion
+
+        #region asb-configure-error-queue
+        class EnableDiagnosticsOnStartup : FunctionsStartup
+        {
+            public override void Configure(IFunctionsHostBuilder builder)
+            {
+                builder.UseNServiceBus(() =>
+                {
+                    var configuration = new ServiceBusTriggeredEndpointConfiguration("MyFunctionsEndpoint");
+                    configuration.AdvancedConfiguration.SendFailedMessagesTo("error");
+                    return configuration;
+                });
+            }
+        }
+        #endregion
+
+        #region asb-enable-diagnostics
+        class configureErrorQueueuOnStartup : FunctionsStartup
+        {
+            public override void Configure(IFunctionsHostBuilder builder)
```

Comment:
could we move the region before the `Configure` methods to leave out the class definition on the snippets?